python/src/cloudstorage/cloudstorage_api.py

```diff
@@ -273,6 +277,184 @@ def listbucket(path_prefix, marker=None, prefix=None, max_keys=None,
+                                      list_of_files,
+                                      files_metadata, 32)
+  compose_object(file_list, destination_file, content_type)
+
+def _compose_batch(list_of_files, destination_file, files_metadata=None,
+            content_type=None, retry_params=None, _account_id=None):
+  """
+  Runs the GCS Compose on the inputed files in batches of 32.
+    Merges between 2 and 1024 files into one file. Composite files may even
+      be built from other existing composites, provided that the total
+      component count does not exceed 1024. See here for details:
+      https://cloud.google.com/storage/docs/composite-objects
+    Automatically breaks down the files into batches of 32.
+    This method is slower and could result in orphan files if it fails.
+  Args:
+    See compose for details
+  Raises:
+    ValueError: If the number of files is outside the range of 2-1024.
+  """
+
+  temp_file_suffix = '____MergeTempFile'
+
+  _, bucket = _validate_compose_list( destination_file,
+                                      list_of_files,
+                                      files_metadata, 1024)
+  """
+  Compose can only handle 32 files at a time. Breaks down the list into batches of 32
+  (this will only need to happen once, since the file_list size restriction is 1024 = 32 * 32).
+  """
+  temp_list = []  # temporary storage for the filenames that store the merged segments of 32
+  if len(list_of_files) > 32:
+    temp_file_counter = 0
+    segments_list = [list_of_files[i:i + 32] for i in range(0, len(list_of_files), 32)]
+    meta_segments_list = [files_metadata[i:i + 32] for i in range(0, len(files_metadata), 32)]
+    files_metadata = []
+    list_of_files = []
+    for segment in segments_list:
+      if len(segment) > 1:
+        temp_file_name = destination_file + temp_file_suffix + str(temp_file_counter)
+        compose(segment, temp_file_name,
+                files_metadata=meta_segments_list,
+                content_type=content_type,
+                retry_params=retry_params,
+                _account_id=_account_id)
+        list_of_files.append(temp_file_name.replace(bucket, '', 1))
+        temp_file_counter += 1
+        temp_list.append(temp_file_name)
+      else:
+        list_of_files.append(segment[0])
+  # There will always be 32 or less files to merge at this point
+  compose(list_of_files, destination_file,
+                files_metadata=files_metadata,
+                content_type=content_type,
+                retry_params=retry_params,
+                _account_id=_account_id)
+  # grab all temp files that were created during the merging of segments of 32
+  temp_list = listbucket(destination_file + temp_file_suffix)
+  # delete all the now-unneeded temporary merge-files for the segments of 32 (if any)
+  for item in temp_list:
+    try:
+      delete(item.filename)
+    except errors.NotFoundError:
+      pass
+
+def _file_exists(destination):
+  """Checks if a file exists.
```

Comment:
method looks good. docstring should change to:
"""Checks if a file exists.
Tries to open the file.
If it succeeds returns True otherwise False.

Args:
  destination: Full path to the file (ie. /bucket/object) with leading slash.

Returns:
  True if the file is accessible otherwise False.
"""